h3/frontend/src/main/java/com/course/servicemesh/frontend/courseservicemeshfrontend/controllers/DashboardController.java

```diff
@@ -0,0 +1,60 @@
+package com.course.servicemesh.frontend.courseservicemeshfrontend.controllers;
+
+import org.apache.tomcat.util.json.JSONParser;
+import org.apache.tomcat.util.json.ParseException;
+import org.asynchttpclient.AsyncHttpClient;
+import org.asynchttpclient.DefaultAsyncHttpClientConfig;
+import org.asynchttpclient.Dsl;
+import org.asynchttpclient.ListenableFuture;
+import org.asynchttpclient.Request;
+import org.asynchttpclient.RequestBuilder;
+import org.asynchttpclient.Response;
+import org.asynchttpclient.util.HttpConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.StringUtils;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.Hashtable;
+import java.util.concurrent.ExecutionException;
+
+@RestController
+@RequestMapping("api/v1/dashboard")
+public class DashboardController {
+
+    private final static Logger logger = LoggerFactory.getLogger(DashboardController.class);
+
+    @GetMapping
+    public Hashtable<String, Object> getDashboard(
+        @RequestHeader(value = "developer", required = false) String devHeader
```

Comment:
This is not correct, a check for "developer" header should be on the ISTIO side. 
You modified the code, to check feature flag, this is incorrect.

Reply:
Moved this to a v2 of frontend server. Now frontend v2 is able to forward "developer" header to author service (where everything else is taken care of by istio's virtual service).